### ISSUE TITLE
Ability to save all visualizations

### ISSUE THREAD

**Issue by maddy-tod** [Qiskit/qiskit-terra#2610]
### What is the expected enhancement?
Currently, any circuit diagrams that are generated can be saved to a file by providing a value to the `filename` argument. It would be good if this was the case for all visualizations, especially the counts visualization.

This is available also (not via an argument but via a separate method) in the Bloch sphere drawer.

**Comment by mtreinish** [comment]
This was discussed during https://github.com/Qiskit/qiskit-terra/issues/1260 when we standardized the visualization functions as part of declaring the visualization interfaces stable. We decided against using a  parameter for this and to just rely on the matplotlib native constructs for writing a file. (just call `.savefig(filename)` on the output Figure object from the function) I did write a tutorial which explained how to do this in: https://github.com/Qiskit/qiskit-tutorials but I can't find it anymore, it might have been deleted during some refactor over there.

As for the circuit visualizers, the reason we left the filename arg there is because all 4 outputs have different types returned by `circuit_drawer()` each with a different way to convert that into a file on disk. So to ensure there was a common method available we kept the filename arg.
